oauth-common/src/main/java/io/strimzi/kafka/oauth/metrics/MetricKey.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2017-2021, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.oauth.metrics;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class MetricKey {
+
+    private static final Set<String> QUOTED_ATTRS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList("host", "path")));
+    private static final List<String> SORTED_ATTRS = Arrays.asList("context", "kind", "host", "path", "mechanism", "outcome", "error_type", "status");
+
+    final String name;
+    final Map<String, String> attrs;
+    final String mbeanName;
+
+    public MetricKey(String name, Map<String, String> attrs) {
+        this.name = name;
+        this.attrs = Collections.unmodifiableMap(attrs);
+        this.mbeanName = composeMBeanName(name, attrs);
+    }
+
+    public static MetricKey of(String name, Map<String, String> attrs) {
+        return new MetricKey(name, attrs);
+    }
+
+    public static MetricKey of(String name, String... attrs) {
+        if (attrs.length == 0) {
+            throw new IllegalArgumentException("There should be some attrs");
+        }
+        if (attrs.length % 2 != 0) {
+            throw new IllegalArgumentException("There should be an even number of attrs");
+        }
+
+        Map<String, String> attrMap = new LinkedHashMap<>();
+        for (int i = 0; i < attrs.length; i += 2) {
+            attrMap.put(attrs[i], attrs[i + 1]);
+        }
+        return new MetricKey(name, attrMap);
+    }
+
+    public String getMBeanName() {
+        return mbeanName;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Map<String, String> getAttrs() {
+        return attrs;
+    }
+
+    private String composeMBeanName(String name, Map<String, String> attrs) {
+        StringBuilder objName = new StringBuilder("strimzi.oauth:").append("type=").append(name);
+        Map<String, String> allAttrs = new HashMap<>(attrs);
+
+        for (String attr: SORTED_ATTRS) {
```

Comment:
A comment about why we need this special sorting semantics.

Reply:
Actually, as a result of the transition to Kafka Metrics, this code, while producing a valid MBean ObjectName, is not used for the ObjectName any more, just to produce a unique Sensor key that incorporates a base name and all the attributes. We could concatenate a name and a string representation of the attributes map, and it would be a unique key just as well. I'll clean this up.